File: src/main/java/com/zoarial/iot/threads/tcp/networkModel/TCPStart.java
package com.zoarial.iot.threads.tcp.networkModel;

import me.zoarial.networkArbiter.annotations.ZoarialObjectElement;
import me.zoarial.networkArbiter.annotations.ZoarialNetworkObject;

import java.util.Objects;

@ZoarialNetworkObject
public class TCPStart {
    @ZoarialObjectElement(placement = 1)
    byte version;

    @ZoarialObjectElement(placement = 2)
    int sessionId;

    @ZoarialObjectElement(placement = 3)
    String request = new String();

    public byte getVersion() {
        return version;
    }

    public int getSessionId() {
        return sessionId;
    }

    public String getRequest() {
        return request;
    }

    public TCPStart(byte version, int sessionId, String request) {
        this.version = version;
        this.sessionId = sessionId;
        this.request = request;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TCPStart tcpStart = (TCPStart) o;
        return version == tcpStart.version && sessionId == tcpStart.sessionId && request.equals(tcpStart.request);
    }

    @Override
    public int hashCode() {
        return Objects.hash(version, sessionId, request);
    }
}

File: src/main/java/com/zoarial/iot/threads/tcp/networkModel/TCPStart.kt
package com.zoarial.iot.threads.tcp.networkModel

import me.zoarial.networkArbiter.annotations.ZoarialNetworkObject
import me.zoarial.networkArbiter.annotations.ZoarialObjectElement
import java.util.*

@ZoarialNetworkObject
class TCPStart(@field:ZoarialObjectElement(placement = 1) var version: Byte, @field:ZoarialObjectElement(placement = 2) var sessionId: Int, request: String) {

    @ZoarialObjectElement(placement = 3)
    var request = String()

    init {
        this.request = request
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val tcpStart = other as TCPStart
        return version == tcpStart.version && sessionId == tcpStart.sessionId && request == tcpStart.request
    }

    override fun hashCode(): Int {
        return Objects.hash(version, sessionId, request)
    }
}